src/glance.gleam

```diff
@@ -1708,6 +1720,16 @@ fn optional_variant_fields(
   }
 }
 
+fn field_pun(tokens: Tokens) -> Result(#(Field(t), Tokens), Error) {
+  case tokens {
+    [#(t.Name(name), _), #(t.Colon, _), #(t.RightParen, _) as end, ..tokens]
+    | [#(t.Name(name), _), #(t.Colon, _), #(t.Comma, _) as end, ..tokens] -> {
+      Ok(#(PunnedField(name), [end, ..tokens]))
+    }
+    _ -> unexpected_error(tokens)
+  }
+}
```

Comment:
Instead of adding a new one, why not update the `field` function directly to take punning into account? Otherwise one would have to remember to also call this every time the `field` function is used. Or is it used in places where there can be no punning as well?

Reply:
I just went through and checked all the places `field` is called and accidentally discovered that I myself already forgot to add `field_pun` to the `fn_capture()` parser haha, good catch!

At first it seemed much easier to add a new function, but in hindsight I don't think there's much reason to.

There is a case in `optional_variant_fields()` (seems to be used for parsing record constructor definitions) where changing `field()` itself could allow something silly like 
```gleam
pub type Wibble {
  Wobble(int:)
}
```
But the README does say that `glance` will parse some things the compiler would reject as invalid iirc, so that might be okay?

Reply:
Mmh I'd try and make sure this is not parsed but let's hear what Louis thinks

Reply:
Good catch. Could we have a test to make sure that this doesn't parse? Thank you

Reply:
I'll get on that on my next break!

Reply:
Done! ([451f1e1](https://github.com/lpil/glance/pull/20/commits/451f1e154e297c6750825442fd9d0c25bac0d34d))